include/kernel.h

```diff
@@ -5610,6 +5611,69 @@ int k_thread_runtime_stats_all_get(k_thread_runtime_stats_t *stats);
 
 #endif
 
+#ifdef CONFIG_DEMAND_PAGING_STATS
```

Comment:
do we need to have this in kernel.h? how about include/sys/mem_manage.h where all other demand paging related APIs are?

Reply:
Putting it in `include/sys/mem_manage.h` would result in circular include because of `k_tid_t` being used by these functions. `k_tid_t` is defined at the end in `kernel/thread.h` which needs to include `include/sys/mem_manage.h` (before struct k_thread) because of the struct to store stat.

Reply:
Let me see how to massage it to work by using `struct k_thread` instead of `k_tid_t`.

Reply:
Using `struct k_thread` works.